test/functional/actions/sendEvent.spec.js

```diff
@@ -68,7 +76,31 @@ test("initializes form fields with full settings", async () => {
   await radioGroup.dataElement.expectChecked();
   await radioGroup.values.expectUnchecked();
   await scopeDataElementField.expectValue("%myDecisionScope%");
-  await scopeValuesFirstField.expectValue("");
+});
+
+test("initializes form fields with full settings, when decision scopes is an array of scopes", async () => {
+  await extensionViewController.init({
+    extensionSettings: mockExtensionSettings,
+    settings: {
+      instanceName: "alloy2",
+      renderDecisions: true,
+      xdm: "%myDataLayer%",
+      type: "myType1",
+      mergeId: "%myMergeId%",
+      decisionScopes: ["foo1", "foo2", "foo3"]
+    }
+  });
+  await instanceNameField.expectValue("alloy2");
+  await renderDecisionsField.expectChecked();
+  await xdmField.expectValue("%myDataLayer%");
+  await typeField.expectValue("myType1");
+  await mergeIdField.expectValue("%myMergeId%");
+  await radioGroup.dataElement.expectUnchecked();
+  await radioGroup.values.expectChecked();
+  await scopeDataElementField.expectError;
+  await scopeArrayValues[0].value.expectValue("foo1");
+  await scopeArrayValues[1].value.expectValue("foo2");
+  await scopeArrayValues[2].value.expectValue("foo3");
```

Comment:
I'm concerned about how much we're duplicating across tests and and how that will compound as more and more options and variations of those options are added. There are several ways to slice this, but I would probably take one approach (say, using a data element) in the "initializes form fields with full settings" test and then make a different test that specifically tests scope array values, like "initializes form fields with scope array values" that doesn't test all the other fields. I'd then take the same approach with "returns full valid settings" and "returns valid settings with scope array values".

Reply:
thank you, I have changed this!